src/ripple/app/main/Application.cpp

```diff
@@ -862,39 +861,52 @@ class ApplicationImp
 
     //------------------------------------------------------------------------------
 
-    void onDeadlineTimer (DeadlineTimer& timer) override
+    void setSweepTimer ()
     {
-        if (timer == m_entropyTimer)
-        {
-            crypto_prng().mix_entropy ();
-            return;
-        }
-
-        if (timer == m_sweepTimer)
-        {
-            // VFALCO TODO Move all this into doSweep
-
-            if (! config_->standalone())
+        sweepTimer_.expires_from_now (
+            std::chrono::seconds {config_->getSize (siSweepInterval)});
+        sweepTimer_.async_wait ([this] (boost::system::error_code const& e)
```

Comment:
Regarding the `strand`, probably so.  You are the `asio` master.  I'll need to read up on `strand`s.